lnwire/features.go

```diff
@@ -155,6 +155,11 @@ const (
 	// TODO: Decide on actual feature bit value.
 	ExplicitChannelTypeOptional = 45
 
+	// PaymentMetadataRequired is a required bit that denotes that an
+	// invoice contains metadata that must be passed along with the payment
+	// htlc(s).
+	PaymentMetadataRequired = 48
```

Comment:
I saw there was some [discussion](https://github.com/lightning/bolts/pull/912#discussion_r717313582) in the spec PR about this, seems like the conclusion was to add the optional feature bit as well? Even if we're not going to use it, might be worth adding it here? (more confusing to not have it, than have it and not use it IMO)

Reply:
Added the optional feature bit.

The Eclair team is going to use this to get an idea of the proportion of senders that have upgraded to node software that supports metadata sending. They want to include some metadata with every invoice generated and then observe the incoming htlcs to see if it is echoed in tlv.

This is something for us to decide on too. Do we want to increase the size of every invoice slightly to include the marker metadat?

Reply:
I think its clearer to just have it there, rather than explain why it's not there. 

> This is something for us to decide on too. Do we want to increase the size of every invoice slightly to include the marker metadata and the optional bit?

Wouldn't this decision fall on implementors rather than lnd itself?

Reply:
With this PR as is, there is no way they can add that marker metadata.